Grunt issues

```
root@mnzt:~/stretto# grunt
Loading "grunt-bower-install-simple.js" tasks...ERROR
>> Error: Cannot find module 'q'
Warning: Task "bower-install-simple:def" not found. Use --force to continue.

Aborted due to warnings.
```

From a fresh git clone. 

Other modules missing are; `mout, configstore, archy, semver-utils, stringify-object, semver-utils, graceful-fs, handlebars`.

After install all these modules with `npm i [module]` I'm then greeted with the following Grunt error:
```
root@gopee:~/stretto# grunt
Running "bower-install-simple:def" (bower-install-simple) task
Executing Bower (Command: install)
Warning: Cannot find module 'glob' Use --force to continue.

Aborted due to warnings.
```

This also throws this error for the following packages; `rimraf, chmodr, md5-hex, retry, lru-cache, lockfile, semver, fstream, fstream-ignore, which, tmp, p-throttler, request-progress, fs-write-stream-atomic, destroy, decompress-zip, tar-fs, shell-quote` After manually installing all of these packages, grunt runs and resolves most assets, before hitting another snag:

```
bower error         cmd(...).spread is not a function

Stack trace:
TypeError: cmd(...).spread is not a function
    at Function.GitRemoteResolver.refs (/root/stretto/node_modules/grunt-bower-i                        nstall-simple/node_modules/bower/lib/core/resolvers/GitRemoteResolver.js:255:6)
    at Function.GitResolver.tags (/root/stretto/node_modules/grunt-bower-install                        -simple/node_modules/bower/lib/core/resolvers/GitResolver.js:297:18)
    at Function.GitResolver.versions (/root/stretto/node_modules/grunt-bower-ins                        tall-simple/node_modules/bower/lib/core/resolvers/GitResolver.js:257:18)
    at GitHubResolver.GitResolver._findResolution (/root/stretto/node_modules/gr                        unt-bower-install-simple/node_modules/bower/lib/core/resolvers/GitResolver.js:10                        6:21)
    at GitHubResolver.GitRemoteResolver._findResolution (/root/stretto/node_modu                        les/grunt-bower-install-simple/node_modules/bower/lib/core/resolvers/GitRemoteRe                        solver.js:99:50)
    at GitHubResolver.GitResolver._resolve (/root/stretto/node_modules/grunt-bow                        er-install-simple/node_modules/bower/lib/core/resolvers/GitResolver.js:65:17)
    at _fulfilled (/root/stretto/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/root/stretto/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/root/stretto/node_modules/q/q.js:796:13                        )
    at /root/stretto/node_modules/q/q.js:604:44

Console trace:
Error
    at StandardRenderer.error (/root/stretto/node_modules/grunt-bower-install-si                        mple/node_modules/bower/lib/renderers/StandardRenderer.js:83:37)
    at Logger.<anonymous> (/root/stretto/node_modules/grunt-bower-install-simple                        /tasks/grunt-bower-install-simple.js:119:22)
    at emitOne (events.js:77:13)
    at Logger.emit (events.js:169:7)
    at Logger.emit (/root/stretto/node_modules/grunt-bower-install-simple/node_m                        odules/bower/node_modules/bower-logger/lib/Logger.js:29:39)
    at /root/stretto/node_modules/grunt-bower-install-simple/node_modules/bower/                        lib/commands/index.js:48:20
    at _rejected (/root/stretto/node_modules/q/q.js:844:24)
    at /root/stretto/node_modules/q/q.js:870:30
    at Promise.when (/root/stretto/node_modules/q/q.js:1122:31)
    at Promise.promise.promiseDispatch (/root/stretto/node_modules/q/q.js:788:41                        )
System info:
Bower version: 1.7.2
Node version: 4.2.4
OS: Linux 3.13.0-68-generic x64
Warning: Task "bower-install-simple:def" failed. Use --force to continue.
``` 

This is most likely due to a version error/incompatibility issue from manually installing the modules.

Did you run a plain `npm install` before grunt? Because you need to do that
and it will automatically install all those packages for you.

On Thu, Jan 7, 2016, 10:00 PM Toby Archer <notifications@github.com> wrote:

> root@mnzt:~/stretto# grunt
> Loading "grunt-bower-install-simple.js" tasks...ERROR
> >> Error: Cannot find module 'q'
> Warning: Task "bower-install-simple:def" not found. Use --force to continue.
>
> Aborted due to warnings.
>
> From a fresh git clone.
>
> Other modules missing are; mout, configstore, archy, semver-utils,
> stringify-object, semver-utils, graceful-fs, handlebars.
>
> After install all these modules with npm i [module] I'm then greeted with
> the following Grunt error:
>
> root@gopee:~/stretto# grunt
> Running "bower-install-simple:def" (bower-install-simple) task
> Executing Bower (Command: install)
> Warning: Cannot find module 'glob' Use --force to continue.
>
> Aborted due to warnings.
>
> This also throws this error for the following packages; rimraf, chmodr,
> md5-hex, retry, lru-cache, lockfile, semver, fstream, fstream-ignore,
> which, tmp, p-throttler, request-progress, fs-write-stream-atomic, destroy,
> decompress-zip, tar-fs, shell-quote After manually installing all of
> these packages, grunt runs and resolves most assets, before hitting another
> snag:
>
> bower error         cmd(...).spread is not a function
>
> Stack trace:
> TypeError: cmd(...).spread is not a function
>     at Function.GitRemoteResolver.refs (/root/stretto/node_modules/grunt-bower-i                        nstall-simple/node_modules/bower/lib/core/resolvers/GitRemoteResolver.js:255:6)
>     at Function.GitResolver.tags (/root/stretto/node_modules/grunt-bower-install                        -simple/node_modules/bower/lib/core/resolvers/GitResolver.js:297:18)
>     at Function.GitResolver.versions (/root/stretto/node_modules/grunt-bower-ins                        tall-simple/node_modules/bower/lib/core/resolvers/GitResolver.js:257:18)
>     at GitHubResolver.GitResolver._findResolution (/root/stretto/node_modules/gr                        unt-bower-install-simple/node_modules/bower/lib/core/resolvers/GitResolver.js:10                        6:21)
>     at GitHubResolver.GitRemoteResolver._findResolution (/root/stretto/node_modu                        les/grunt-bower-install-simple/node_modules/bower/lib/core/resolvers/GitRemoteRe                        solver.js:99:50)
>     at GitHubResolver.GitResolver._resolve (/root/stretto/node_modules/grunt-bow                        er-install-simple/node_modules/bower/lib/core/resolvers/GitResolver.js:65:17)
>     at _fulfilled (/root/stretto/node_modules/q/q.js:834:54)
>     at self.promiseDispatch.done (/root/stretto/node_modules/q/q.js:863:30)
>     at Promise.promise.promiseDispatch (/root/stretto/node_modules/q/q.js:796:13                        )
>     at /root/stretto/node_modules/q/q.js:604:44
>
> Console trace:
> Error
>     at StandardRenderer.error (/root/stretto/node_modules/grunt-bower-install-si                        mple/node_modules/bower/lib/renderers/StandardRenderer.js:83:37)
>     at Logger.<anonymous> (/root/stretto/node_modules/grunt-bower-install-simple                        /tasks/grunt-bower-install-simple.js:119:22)
>     at emitOne (events.js:77:13)
>     at Logger.emit (events.js:169:7)
>     at Logger.emit (/root/stretto/node_modules/grunt-bower-install-simple/node_m                        odules/bower/node_modules/bower-logger/lib/Logger.js:29:39)
>     at /root/stretto/node_modules/grunt-bower-install-simple/node_modules/bower/                        lib/commands/index.js:48:20
>     at _rejected (/root/stretto/node_modules/q/q.js:844:24)
>     at /root/stretto/node_modules/q/q.js:870:30
>     at Promise.when (/root/stretto/node_modules/q/q.js:1122:31)
>     at Promise.promise.promiseDispatch (/root/stretto/node_modules/q/q.js:788:41                        )
> System info:
> Bower version: 1.7.2
> Node version: 4.2.4
> OS: Linux 3.13.0-68-generic x64
> Warning: Task "bower-install-simple:def" failed. Use --force to continue.
>
> This is most likely due to a version error/incompatibility issue from
> manually installing the modules.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/benkaiser/stretto/issues/83>.
>

--- a/README.md
+++ b/README.md
@@ -20,7 +20,7 @@ Visit the [releases page](https://github.com/benkaiser/stretto/releases) for the
 
 ### Dev Installation Instructions
 ##### Dependencies
-[node](http://nodejs.org/), [git](http://git-scm.com/)
+[node](http://nodejs.org/), [git](http://git-scm.com/), [npm v3+](http://npmjs.org)
 
 ##### NPM Global Dependencies
 [grunt-cli](http://gruntjs.com/getting-started#installing-the-cli), [nodemon](http://nodemon.io/)
@@ -45,7 +45,12 @@ Then run the server with:
 ```
 node app.js
 ```
-Go to `localhost:2000` in your browser (or known ip of server if it is on a different machine). From there a prompt will show so you can change your music directory, after saving your music directory you can click `Scan Library -> Regular Scan` at the top of the page. From there select `Start Scan` and let it do it's magic adding all the songs to your library. Happy listening!
+
+Alternatively, install PM2 (read more about PM2 at [pm2.keymetrics.io](http://http://pm2.keymetrics.io/))with `npm install -g pm2` and run the server persistently with:
+```
+pm2 start app.js
+```
+Go to `localhost:2000` in your browser (or known IP of server if it is on a different machine). From there a prompt will allow you to change your music directory, after saving your music directory you can click `Scan Library -> Regular Scan` at the top of the page. From there select `Start Scan` and let it do it's magic adding all the songs to your library. Happy listening!
 
 ### Android Sync App
 
@@ -74,8 +79,8 @@ bindsym $mod+Prior exec curl http://localhost:2000/command/my_remote_name/next
 #### Defining Features
 
 - Sync between computers (share your library with your friends!)
-- Generate mix from songs streamed from youtube, allowing you to explore new music from your current library
-- Download from youtube and soundcloud
+- Generate mix from songs streamed from YouTube, allowing you to explore new music from your current library
+- Download from YouTube and SoundCloud
 - Built in the open with other open source libraries
 
 #### Simple Features

